apmproxy/receiver.go

```diff
@@ -87,9 +93,39 @@ func (c *Client) handleInfoRequest() (func(w http.ResponseWriter, r *http.Reques
 	customTransport.ResponseHeaderTimeout = c.client.Timeout
 	reverseProxy.Transport = customTransport
 
+	reverseProxy.ModifyResponse = func(rsp *http.Response) error {
+		if rsp.StatusCode != http.StatusOK {
+			b, err := io.ReadAll(rsp.Body)
+			if err != nil {
+				c.logger.Warnf("failed to read version response body: %v", err)
+				b = []byte{}
```

Comment:
Is there any harm in using the result of `io.ReadAll` below? It's likely to be nil, which is fine to type-assert to a string. If it's not nil, it may be interesting to log that.

Reply:
The `ModifyResponse` func has been removed so this doesn't apply anymore.